daml-lf/encoder/src/test/lf/Enum_all_.lf

```diff
@@ -32,6 +32,19 @@ module EnumMod {
     right: EnumMod:ColoredTree
   };
 
+  record @serializable TestTemplatesRecord = {
+    optionalColor: EnumMod:OptionalColor,
+    coloredTree: EnumMod:ColoredTree,
+    party: Party
+  };
+
+  template (this : TestTemplatesRecord) =  {
+    precondition True;
+    signatories (PartyMod:one (EnumMod:TestTemplatesRecord {party} this));
+    observers (Nil @Party);
+    agreement "Agreement for EnumMod:TestTemplatesRecord";
+  };
+
```

Comment:
@remyhaemmerle-da This should be the only work that impacts the `daml-lf` codebase, let me know if that's ok with you.

Reply:
Perfectly fine for me.
I would just have prefix the new definition with `Test`, as everything here is for testing something. 
Maybe something like `NestedEnumTemplated` .

Reply:
Addressed by https://github.com/digital-asset/daml/pull/13571/commits/038840c740f8234860045de8f59cf1776f5663cf in https://github.com/digital-asset/daml/pull/13571